### ISSUE TITLE
syncMoreHistoricMessages failing

### ISSUE THREAD

**Issue by rajohns08** [layerhq/Atlas-Android#211]
`User B` created a conversation with `User A`. 

When `User A` does:
```
Conversation conversation;
try {
    conversation = App.layerClient.newConversationWithUserIds(participantId);
} catch (LayerConversationException e) {
    conversation = e.getConversation();
}
conversation.syncMoreHistoricMessages(HISTORIC_MESSAGES_PER_FETCH);
```
everything works fine.

When `User B` tries to use the above method for getting the conversation with `User A`, I get the following:

> Ignoring call to `syncMoreHistoricMessages`. Cannot request `syncMoreHistoricMessages` on a new Conversation: ConversationImpl [  id: layer:///conversations/a338b9df-808c-497f-b1fb-1aeb0352c7a2, mStreamDatabaseId: null, mStreamId: null, mCreatedAt: null, mDatabaseId: null, mDistinct: true, mTotalMessageCount: null, mTotalUnreadMessageCount: null, mHistoricSyncStatus: INVALID, mReadReceiptsEnabled: true, mParticipants: [0: 1 : participant [  mMemberId: 1, mDatabaseId: null, mConversationDatabaseId: null, mStreamMemberDatabaseId: null], 1: 2 : participant [  mMemberId: 2, mDatabaseId: null, mConversationDatabaseId: null, mStreamMemberDatabaseId: null]], mMetadata: MetadataImpl{mMap={}}]

yet it still successfully loads the last message between the two users.

**Comment by rajohns08** [comment]
Also note if I instantiate my `LayerClient` with `.historicSyncPolicy(LayerClient.Options.HistoricSyncPolicy.ALL_MESSAGES)` it DOES successfully sync all messages.

**Comment by csbrandt** [comment]
For technical assistance please reach out to us via [support.layer.com](https://support.layer.com).

Thanks,
Chris

**Comment by rajohns08** [comment]
I think the heart of this issue was that when I was calling `newConversationWithUserIds`, Layer hadn't fully sync'd all conversations yet. It later corrected itself, but after I had already called `syncMoreHistoricMessages`.

I had to hack a way around this issue by listening for Layer change events. By the time I get a `LayerObject.Type.CONVERSATION` change event, I know Layer is fully synced so I call `syncMoreHistoricMessages`.

I don't know of another way to wait on all conversation history to sync before I call `newConversationWithUserIds` and have it fail as I mentioned in my first post.
